pkg/genericapiserver/enablement_config.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+type APIResourceEnablement interface {
+	ResourceEnabled(resource unversioned.GroupVersionResource) bool
+	VersionEnabled(version unversioned.GroupVersion) bool
+}
+
+// Specifies the overrides for various API group versions.
+// This can be used to enable/disable entire group versions or specific resources.
+type APIGroupVersionEnablementConfig struct {
+	// Whether to enable or disable this entire group version.
+	Disable bool
+
+	// DisabledResources lists the resources that are specifically disabled for a group/version
+	// DisabledResources trumps EnabledResources
+	DisabledResources sets.String
+
+	// EnabledResources lists the resources that should be enabled by default.  This is a little
+	// unusual, but we need it for compatibility with old code for now.  An empty set means
+	// enable all.
+	EnabledResources sets.String
+}
+
+type APIResourceEnablementConfig struct {
+	GroupVersionDisablers map[unversioned.GroupVersion]*APIGroupVersionEnablementConfig
+}
+
```

Comment:
Add `var _ APIResourceEnablement = &APIResourceEnablementConfig{}` to ensure that this implements the interface.